jaeger_client/metrics.py

```diff
@@ -21,69 +21,143 @@
 from __future__ import absolute_import
 
 
-class Metrics(object):
-    """Provides an abstraction of metrics reporting framework."""
+class MetricsFactory(object):
+    """Generates new metrics."""
 
-    # TODO do we want to automatically include host name?
-    prefix = 'jaeger'
+    def __init__(self, count=None, gauge=None, timing=None, tags=None):
+        """
+        :param count: function to emit counters
+        :param gauge: function to emit gauges
+        :param timing: function to emit timers
+        :param tags: global {k:v} dictionary
+        """
+        self._count = count
+        self._gauge = gauge
+        self._timing = timing
+        self._tags = tags
+        if not callable(self._count):
+            self._count = self._noop
+        if not callable(self._gauge):
+            self._gauge = self._noop
+        if not callable(self._timing):
+            self._timing = self._noop
 
-    TRACES_STARTED_SAMPLED = '%s.traces-started.sampled' % prefix
-    """Number of traces started by this tracer as sampled"""
+    def _noop(self, *args):
+        pass
 
-    TRACES_STARTED_NOT_SAMPLED = '%s.traces-started.not-sampled' % prefix
-    """Number of traces started by this tracer as not sampled"""
+    def create_counter(self, name, tags=None):
+        """
+        Generates a new counter from the given name and tags and returns
+        a callable function used to increment the counter.
+        :param name: name of the counter
+        :param tags: tags for the counter
+        :return: a callable function which takes the value to increase
+        the counter by ie. def increment(value)
+        """
+        def increment(value):
+            return self._noop()
+        return increment
```

Comment:
why can't this be simply `return self._noop`? And if it can, can't we just throw away all of the constructor code?  Then in the Legacy class, instead of populating these fields, just store `self._metrics` class and use it directly in `create_timer` and such.